src/doc/src/reference/semver.md

```diff
@@ -206,6 +208,111 @@ This is not considered a major change because conventionally glob imports are
 a known forwards-compatibility hazard. Glob imports of items from external
 crates should be avoided.
 
+<a id="type-layout"></a>
+### Major: Changing the alignment, layout, or size of a well-defined type
+
+It is a breaking change to change the alignment, layout, or size of a type that was previously well-defined.
+
+In general, nominal types that use the [the default representation] do not have a well-defined alignment, layout, or size.
+The compiler is free to alter the alignment or layout, so code should not make any assumptions about it.
+
+> **Note**: It may be possible for external crates to break if they make assumptions about the alignment, layout, or size of a type even if it is not well-defined.
+> This is not considered a SemVer breaking change since those assumptions should not be made.
+
+Some examples of changes that are not a breaking change are (assuming no other rules in this guide are violated):
+
+* Adding, removing, or changing fields of a default representation struct, union, or enum.
+* Adding variants to a default representation enum.
```

Comment:
Likewise, this one is *often* going to be breaking, in the absence of `non_exhaustive` or an existing private variant. I don't want us to redundantly cover potentially breaking changes *repeatedly*, but I feel like this could use a bit of qualification (e.g. "Adding variants to a default representation enum, if the enum already has private variants or uses `non_exhaustive`")

Reply:
Added qualification here, too, with links.